src/Lyn.Protocol/Bolt2/ChannelEstablishment/OpenChannelMessageService.cs

```diff
@@ -137,14 +127,93 @@ public async Task ProcessMessageAsync(PeerMessage<OpenChannel> message)
+            _channelStateRepository.Create(channelState);
 
             await _messageSender.SendMessageAsync(new PeerMessage<AcceptChannel>(message.NodeId, acceptChannel));
         }
+
+        private string CheckMessage(OpenChannel openChannel, ChainParameters chainParameters, ChannelConfig channelConfig)
+        {
+            // Bolt 2 -
+            //  The receiving node MAY fail the channel if:
+            //    announce_channel is false(0), yet it wishes to publicly announce the channel.
+            //    funding_satoshis is too small.
+            //    it considers htlc_minimum_msat too large.
+            //    it considers max_htlc_value_in_flight_msat too small.
+            //    it considers channel_reserve_satoshis too large.
+            //    it considers max_accepted_htlcs too small.
+            //    it considers dust_limit_satoshis too small and plans to rely on the sending node publishing its commitment
+            //       transaction in the event of a data loss(see message-retransmission).
+
+            if (openChannel.FundingSatoshis < chainParameters.MinFundingAmount) return "funding_satoshis is too small";
+            if (openChannel.HtlcMinimumMsat > channelConfig.HtlcMinimum) return "htlc_minimum_msat too large";
+            if (openChannel.MaxHtlcValueInFlightMsat < channelConfig.MaxHtlcValueInFlight) return "max_htlc_value_in_flight_msat too small";
+            if (openChannel.ChannelReserveSatoshis > channelConfig.ChannelReserve) return "channel_reserve_satoshis too large";
+            if (openChannel.MaxAcceptedHtlcs < channelConfig.MaxAcceptedHtlcs) return "max_accepted_htlcs too small";
+            if (openChannel.DustLimitSatoshis < channelConfig.DustLimit) return "dust_limit_satoshis too small";
+
+            // Bolt 2 -
+            //  The receiving node MUST fail the channel if:
+            //    the chain_hash value is set to a hash of a chain that is unknown to the receiver.
+            //    push_msat is greater than funding_satoshis * 1000.
+            //    to_self_delay is unreasonably large.
+            //    max_accepted_htlcs is greater than 483.
+            //    it considers feerate_per_kw too small for timely processing or unreasonably large.
+            //    funding_pubkey, revocation_basepoint, htlc_basepoint, payment_basepoint, or delayed_payment_basepoint are not valid secp256k1 pubkeys in compressed format.
+            //    dust_limit_satoshis is greater than channel_reserve_satoshis.
+            //    the funder's amount for the initial commitment transaction is not sufficient for full fee payment.
+            //    both to_local and to_remote amounts for the initial commitment transaction are less than or equal to channel_reserve_satoshis(see BOLT 3).
+            //    funding_satoshis is greater than or equal to 2 ^ 24 and the receiver does not support option_support_large_channel.
+
+            MiliSatoshis fundingMiliSatoshis = openChannel.FundingSatoshis;
```

Comment:
I think this could go in another private method for MUST on failing the channel

Reply:
I don't follow, this is just converting to mili satoshis